lib/LocalModel.js

```diff
@@ -210,21 +231,36 @@ LocalModel.prototype.predict = function (inputData, missingStrategy,
   }
 };
 
-LocalModel.prototype.validateInput = function (inputData, cb) {
+LocalModel.prototype.validateInput = function (inputData,
+                                               addUnusedFields,
+                                               cb) {
   /**
    * Validates the syntax of input data.
    *
    * The input fields must be keyed by field name or field id.
    * @param {object} inputData Input data to predict
+   * @param {boolean} addUnusedFields Causes the validation to return the
+   *                                  list of fields in inputData that are
+   *                                  not used
    * @param {function} cb Callback
    */
-  var newInputData = {}, field, inputDataKey;
+  var newInputData = {}, field, inputDataKey, unusedFields = [];
+  // backward compatibility for previous code with no addUnusedFields argument
+  if (typeof addUnusedFields === "function") {
+    cb = addUnusedFields;
+    addUnusedFields = false;
+  }
   if (this.ready) {
     for (field in inputData) {
       if (inputData.hasOwnProperty(field)) {
         if (inputData[field] === null ||
             (typeof this.tree.fields[field] === 'undefined' &&
-             typeof this.invertedFields[field] === 'undefined')) {
+             typeof this.invertedFields[field] === 'undefined') ||
+             (typeof this.tree.fields[field] !== 'undefined' &&
+              this.tree.objectiveField === field) ||
+             (typeof this.invertedFields[field] !== 'undefined' &&
+              this.tree.objectiveField === this.invertedFields[field])) {
```

Comment:
there's a slight difference. The objective field and fields structure are attributes of hte local logistic regression, not of `tree`